### PR TITLE
test: adds make target for deploy manifest

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

**What this PR does / why we need it**:
This PR adds e2e test on deploy manifest for Secrets Store CSI driver. Ref #297 

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -365,6 +370,18 @@ e2e-test: e2e-bootstrap e2e-helm-deploy # run test for windows
 e2e-teardown: $(HELM)
 	helm delete csi-secrets-store --namespace kube-system
 
+.PHONY: e2e-deploy-manifest
+e2e-deploy-manifest:
```
Comment (aramase) [inline Makefile:374]: If you want to test this without the https://github.com/kubernetes/test-infra/pull/23054 changes, you can update this target to `e2e-helm-deploy` and see the results.

File: Makefile [hunk 2]
```diff
@@ -365,6 +370,18 @@ e2e-test: e2e-bootstrap e2e-helm-deploy # run test for windows
...
+	kubectl apply -f manifest_staging/deploy/csidriver.yaml
+	kubectl apply -f manifest_staging/deploy/rbac-secretproviderclass.yaml
+	kubectl apply -f manifest_staging/deploy/rbac-secretproviderrotation.yaml
+	kubectl apply -f manifest_staging/deploy/rbac-secretprovidersyncing.yaml
+	kubectl apply -f manifest_staging/deploy/secrets-store-csi-driver-windows.yaml
```
Comment (aramase) [inline Makefile:379]: Same substitution for windows too

**Review by aramase — COMMENTED** [review]
@nilekhc did you get a chance to validate on prow based on https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/669#discussion_r678723141?

**Comment by nilekhc** [comment]
> @nilekhc did you get a chance to validate on prow based on [#669 (comment)](https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/669#discussion_r678723141)?

@aramase By doing this helm upgrade test is [failing](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_secrets-store-csi-driver/669/pull-secrets-store-csi-driver-e2e-azure/1422305035315843072). Can we merge test-infra PR fiest?

**Comment by aramase** [comment]
> @aramase By doing this helm upgrade test is failing. Can we merge test-infra PR fiest?

There is an outstanding comment in the test-infra PR. Also, I don't see us handling helm test vs yaml tests. So even with the `test-infra` change this PR will still fail because it's trying helm upgrade? There should be a change to skip the `helm upgrade` test if testing yamls.

File: Makefile (hunk 2 above)
Comment (nilekhc) [reply Makefile:379]: We are only building linux image for test. Hence skipping for windows

File: Makefile (hunk 1 above)
Comment (nilekhc) [reply Makefile]: It [works](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_secrets-store-csi-driver/669/pull-secrets-store-csi-driver-e2e-azure/1422620669878734848).

**Review by aramase — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**